src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
```

Comment:
```suggestion
        this[i] = '';
```
Here you are changing the incoming array. The original method `join` does not change the initial array. It should only return a string but the array should be untouched. When you write `this[i] - ''` it means that you make changes to the array (which is not allowed)